pkg/management/postgres/initdb.go

```diff
@@ -341,16 +324,41 @@ func (info InitInfo) ConfigureNewInstance(instance *Instance) error {
 	if err = info.executeSQLRefs(dbTemplate, info.PostInitTemplateSQLRefsFolder); err != nil {
 		return fmt.Errorf("could not execute post init application SQL refs: %w", err)
 	}
+
+	filePath := filepath.Join(info.PgData, constants.CheckEmptyWalArchiveFile)
+	// We create the check empty wal archive file to tell that we should check if the
+	// destination path it is empty
+	if err := fileutils.CreateEmptyFile(filepath.Clean(filePath)); err != nil {
+		return fmt.Errorf("could not create %v file: %w", filePath, err)
+	}
+
+	if info.ApplicationUser == "" {
+		return nil
+	}
+
+	var existsRole bool
+	userRow := dbSuperUser.QueryRow("SELECT COUNT(*) > 0 FROM pg_catalog.pg_roles WHERE rolname = $1",
+		info.ApplicationUser)
+	if err = userRow.Scan(&existsRole); err != nil {
+		return err
+	}
+
+	if !existsRole {
+		if _, err = dbSuperUser.Exec(fmt.Sprintf(
+			"CREATE ROLE %v LOGIN",
+			pgx.Identifier{info.ApplicationUser}.Sanitize())); err != nil {
+			return err
+		}
+	}
```

Comment:
This change could potentially disrupt any initialization scripts that assume the application user is already created.

Reply:
I've created #7870 to address this.